### PR TITLE
test: Add ability to have any number of ganache servers for e2e

### PR DESCRIPTION
**Pull request by tmashuang**
Adds ability to create more than two ganache servers for e2e testing for request queuing.

## **Description**

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/23783?quickstart=1)

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Pre-merge author checklist**

- [ x ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ x ] I've completed the PR template to the best of my ability
- [ x ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ x ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by adonesky1 — APPROVED** [review]
LGTM!

File: test/e2e/helpers.js [hunk 1]
```diff
@@ -76,14 +76,17 @@ async function withFixtures(options, testSuite) {
...
-      const { port, chainId, ganacheOptions2 } = ganacheOptions.concurrent;
-      secondaryGanacheServer = new Ganache();
-      await secondaryGanacheServer.start({
-        blockTime: 2,
-        chain: { chainId },
-        port,
-        vmErrorsOnRPCResponse: false,
-        ...ganacheOptions2,
+      ganacheOptions.concurrent.forEach(async (ganacheSettings) => {
+        const { port, chainId, ganacheOptions2 } = ganacheSettings;
```
Comment (jiexi) [inline test/e2e/helpers.js:80]: i know this was named `ganacheOptions2` prior to your changes, but it seems like an odd naming choice now given that concurrent now accepts an array

File: test/e2e/helpers.js [hunk 2]
```diff
@@ -57,7 +57,7 @@ async function withFixtures(options, testSuite) {
   const bundlerServer = new Bundler();
   const https = await mockttp.generateCACertificate();
   const mockServer = mockttp.getLocal({ https, cors: true });
-  let secondaryGanacheServer;
+  const secondaryGanacheServer = [];
```
Comment (jiexi) [inline test/e2e/helpers.js:60]: wonder if can we combine `ganacheServer` and `secondaryGanacheServer` into a single `ganacheServers`

**Review by jiexi — APPROVED** [review]
As discussed, will not block on refactoring this logic

**Pull request merged**